OpenSim/Common/Property.h

```diff
@@ -938,6 +952,88 @@ readSimplePropertyFromStream(std::istream& in)
        return SimTK::readUnformatted(in, values);
 }
 
+// We provide specializations for all types that contain floats to manage
+// the number of digits displayed. We rely on the default setting of how
+// `stringstream`s handle floats, where `precision` is the number of digits
+// printed, and the `stream` automatically determines whether float or
+// exponential notation is used.
+template <> inline void SimpleProperty<double>::
+writeSimplePropertyToStreamForDisplay(std::ostream& o, const int precision) const
+{
+    o << std::setprecision(precision);
```

Comment:
I wonder if you could do something like the following:

```cpp
o << std::setprecision(precision);
SimTK::writeUnformatted(o, values);
// restore precision
```

Reply:
Oh that's interesting.. hadn't thought of that. If that even works, is it still useful to have all of these specializations on the OpenSim side?

Reply:
Nevermind, if I understand what's going on on the Simbody side, numbers are converted to `String` first using Simbody's `String` constructor before being passed into the stream, so precision should have no effect. This is how it guarantees that sufficient digits are printed for reproducibility.

Reply:
oh, okay. Nevermind then.